### PR TITLE
IncidentsOptions

### PR DESCRIPTION
**Pull request by korshaknn**
### Description

API to set `IncidentsOptions` to configure live incidents

### Changelog

```
<changelog>IncidentsOptions added to NavigationOptions</changelog>
```

### CODE REVIEW

**Review by wanfranck — APPROVED** [review]

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/options/IncidentsOptions.kt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+/**
+ * Electronic Horizon supports live incidents on a most probable path.
+ * To enable live incidents [IncidentsOptions] should be provided.
+ * If both [graph] and [apiUrl] are empty, live incidents are disabled (by default).
+ *
+ * @param graph incidents provider graph name
+ * @param apiUrl LTS incidents service API url, if empty line is supplied will use default url
+ */
+class IncidentsOptions private constructor(
```
Comment (LukasPaczos) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/options/IncidentsOptions.kt:11]: Missing an `IncidentsOptions` test taht extend from `BuilderTest`.

Comment (korshaknn) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/options/IncidentsOptions.kt:11]: fixed

**Comment by korshaknn** [comment]
@LukasPaczos ready to the next round

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/options/NavigationOptions.kt [hunk 2]
```diff
@@ -139,6 +144,7 @@ class NavigationOptions private constructor(
             "deviceProfile=$deviceProfile, " +
             "eHorizonOptions=$eHorizonOptions" +
             "isRouteRefreshEnabled=$isRouteRefreshEnabled" +
+            "incidentsOptions=$incidentsOptions" +
```
Comment (Guardiola31337) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/options/NavigationOptions.kt:147]:
```suggestion
            "eHorizonOptions=$eHorizonOptions, " +
            "isRouteRefreshEnabled=$isRouteRefreshEnabled, " +
            "incidentsOptions=$incidentsOptions" +
```

File: libnavigation-base/src/test/java/com/mapbox/navigation/base/options/IncidentsOptionsTest.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import com.mapbox.navigation.testing.BuilderTest
+import org.junit.Test
+
+class IncidentsOptionsTest : BuilderTest<IncidentsOptions, IncidentsOptions.Builder>() {
+
+    override fun getImplementationClass() = IncidentsOptions::class
+
+    override fun getFilledUpBuilder() = IncidentsOptions.Builder()
+        .graph("graph")
+        .apiUrl("apiUrl")
```
Comment (Guardiola31337) [inline libnavigation-base/src/test/java/com/mapbox/navigation/base/options/IncidentsOptionsTest.kt:12]: Can we use real values to know / understand / document better how these APIs work?

**Review by Guardiola31337 — COMMENTED** [review]
Left a couple of minor comments.

**Review by LukasPaczos — APPROVED** [review]
lgtm

File: libnavigation-base/src/test/java/com/mapbox/navigation/base/options/IncidentsOptionsTest.kt (hunk 3 above)
Comment (korshaknn) [reply libnavigation-base/src/test/java/com/mapbox/navigation/base/options/IncidentsOptionsTest.kt:12]: it's just a Builder test, mock values can be used

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/options/NavigationOptions.kt (hunk 2 above)
Comment (korshaknn) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/options/NavigationOptions.kt:147]: done

**Pull request merged**